src/tokenizer/generic_sql.cpp

```diff
@@ -0,0 +1,201 @@
+            advance(token.str.size() - 1);
+        } else if (!command.empty()) {
+            token.type = sql_token_type::command;
+            token.str = substr(token.index, command.size());
+            advance(token.str.size() - 1);
+        } else if (!ident.empty()) {
+            token.type = sql_token_type::identifier;
+            token.str = substr(token.index, ident.size());
+            advance(token.str.size() - 1);
+        }
+        emplace_token(token);
+    }
+}
+
+void generic_sql_tokenizer::tokenize_inline_comment_or_operator()
+{
+    // The first character is / so it can be a comment or a binary operator
+    sql_token token;
+    token.index = index();
+    if (advance() && peek() == '*') {
+        // Comment
+        advance(); // Skip the '*' in prev()
+        while (advance()) {
+            if (prev() == '*' && peek() == '/') {
+                break;
+            }
+        }
+        token.str = substr(token.index, index() - token.index + 1);
+        token.type = sql_token_type::inline_comment;
+    } else {
+        token.str = substr(token.index, 1);
+        token.type = sql_token_type::binary_operator;
+    }
+    emplace_token(token);
+}
+
+void generic_sql_tokenizer::tokenize_eol_comment()
+{
+    // Inline comment
+    sql_token token;
+    token.index = index();
+
+    while (advance() && peek() != '\n') {}
+
+    token.str = substr(token.index, index() - token.index);
+    token.type = sql_token_type::eol_comment;
+    emplace_token(token);
+}
+
+void generic_sql_tokenizer::tokenize_eol_comment_operator_or_number()
+{
+    if (next() == '-') {
+        tokenize_eol_comment();
+        return;
+    }
+
+    sql_token token;
+    token.index = index();
+
+    auto number_str = extract_number();
+    if (!number_str.empty()) {
+        token.type = sql_token_type::number;
+        token.str = number_str;
+        advance(number_str.size() - 1);
+    } else {
+        // If it's not a number, it must be an operator
+        token.str = substr(token.index, 1);
+        token.type = sql_token_type::binary_operator;
+    }
+
+    emplace_token(token);
+}
+
+void generic_sql_tokenizer::tokenize_operator_or_number()
+{
+    sql_token token;
+    token.index = index();
+
+    auto number_str = extract_number();
+    if (!number_str.empty()) {
+        token.type = sql_token_type::number;
+        token.str = number_str;
+        advance(number_str.size() - 1);
+    } else {
+        // If it's not a number, it must be an operator
+        token.str = substr(token.index, 1);
+        token.type = sql_token_type::binary_operator;
+    }
+
+    emplace_token(token);
+}
+
+std::vector<sql_token> generic_sql_tokenizer::tokenize_impl()
+{
+    for (; !eof(); advance()) {
+        auto c = peek();
+        // TODO use an array of characters or a giant switch?
+        if (ddwaf::isalpha(c) || c == '_' ||
+            static_cast<unsigned char>(c) > 0x7f) { // Command or identifier
```

Comment:
the nomenclature is off: s/command/keyword/ an SQL command is a sequence of tokens terminated by ; or EOF

Reply:
True, I followed the original nomenclature which also didn't feel right.

Reply:
Fixed